src/Sarif.Viewer.VisualStudio/Sarif/Result.extensions.cs

```diff
@@ -42,7 +42,7 @@ public static string GetPrimaryTargetFile(this Result result)
         {
             if (result == null || result.Locations == null || result.Locations.Count == 0)
             {
-                return null;
+                return string.Empty;
```

Comment:
This change was required because the FileName is used as a dictionary key.  Previously this didn't matter because errors without a location were never used in this dictionary.

Reply:
IMO it should still return `null` if `result` is `null`. The error list won't contain a `null`, so we don't have to worry about this being used as a key.

---
In reply to: [135596586](https://github.com/Microsoft/sarif-sdk/pull/600#discussion_r135596586) [](ancestors = 135596586)